core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/Experiment.java

```diff
@@ -12,231 +12,369 @@
 
-  public void setId(long id);
+  private String alias;
 
-  /**
-   * Returns the study of this Experiment object.
-   * 
-   * @return Study study.
-   */
-  public Study getStudy();
+  @OneToMany(targetEntity = ExperimentChangeLog.class, mappedBy = "experiment")
+  private final List<ChangeLog> changeLog = new ArrayList<>();
 
-  /**
-   * Sets the study of this Experiment object.
-   * 
-   * @param s
-   *          study.
-   * 
-   */
-  public void setStudy(Study s);
+  private String description;
 
-  /**
-   * Returns the accession of this Experiment object.
-   * 
-   * @return String accession.
-   */
-  public String getAccession();
+  @Id
+  @GeneratedValue(strategy = GenerationType.AUTO)
+  private long experimentId = UNSAVED_ID;
 
-  /**
-   * Sets the accession of this Experiment object.
-   * 
-   * @param accession
-   *          accession.
-   * 
-   */
-  public void setAccession(String accession);
+  @ManyToMany(targetEntity = KitImpl.class)
+  @JoinTable(name = "Experiment_Kit", inverseJoinColumns = { @JoinColumn(name = "kits_kitId") }, joinColumns = {
+      @JoinColumn(name = "experiments_experimentId") })
+  private Collection<Kit> kits = new HashSet<>();
 
-  /**
-   * Returns the alias of this Experiment object.
-   * 
-   * @return String alias.
-   */
-  public String getAlias();
+  @ManyToOne(targetEntity = UserImpl.class)
+  @JoinColumn(name = "lastModifier")
+  private User lastModifier;
 
-  /**
-   * Sets the alias of this Experiment object.
-   * 
-   * @param alias
-   *          alias.
-   * 
-   */
-  public void setAlias(String alias);
+  // defines a pool on which this experiment will operate. This contains one or more dilutions of a sample
+  @ManyToOne(targetEntity = LibraryImpl.class)
+  @JoinColumn(name = "library_libraryId")
+  private Library library;
 
-  /**
-   * Returns the title of this Experiment object.
-   * 
-   * @return String title.
-   */
-  public String getTitle();
+  @Column(nullable = false)
+  private String name;
 
-  /**
-   * Sets the title of this Experiment object.
-   * 
-   * @param title
-   *          title.
-   * 
-   */
-  public void setTitle(String title);
+  @ManyToOne(targetEntity = Platform.class)
+  @JoinColumn(name = "platform_platformId")
+  private Platform platform;
+
+  // defines the parent run which processes this experiment
+  @OneToMany(mappedBy = "experiment", cascade=CascadeType.ALL)
+  private List<RunPartition> runPartitions;
+
+  @ManyToOne(cascade = CascadeType.PERSIST)
+  @JoinColumn(name = "securityProfile_profileId")
+  private SecurityProfile securityProfile = new SecurityProfile();
+
+  @ManyToOne(targetEntity = StudyImpl.class)
+  @JoinColumn(name = "study_studyId")
+  private Study study;
+  @Column(nullable = false)
+  private String title;
 
   /**
-   * Returns the name of this Experiment object.
-   * 
-   * @return String name.
+   * Construct a new Experiment with a default empty SecurityProfile
    */
+  public Experiment() {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)